src/components/RightSidebar/SearchMessages/SearchMessagesTab.vue

```diff
@@ -0,0 +1,442 @@
+const sinceDate = ref<Date | null>(null)
+const untilDate = ref<Date | null>(null)
+const searchLimit = ref(10)
+const searchCursor = ref(0)
+const searchDetailsOpened = ref(false)
+const isFetchingResults = ref(false)
+const isSearchExhausted = ref(false)
+
+const store = useStore()
+const isInCall = useIsInCall()
+
+const token = computed(() => store.getters.getToken())
+const participantsInitialised = computed(() => store.getters.participantsInitialised(token.value))
+const participants = computed<UserFilterObject>(() => {
+	return store.getters.participantsList(token.value)
+		.map(({ actorId, displayName, actorType }: { actorId: string; displayName: string; actorType: string}) => ({
+			id: actorId,
+			displayName,
+			isNoUser: actorType !== 'users',
+			user: actorId,
+			disableMenu: true,
+			showUserStatus: false,
+		}))
+})
+const canLoadMore = computed(() => !isSearchExhausted.value && !isFetchingResults.value && searchCursor.value !== 0)
+const hasFilter = computed(() => fromUser.value || sinceDate.value || untilDate.value)
+
+onMounted(() => {
+	EventBus.on('route-change', onRouteChange)
+})
+
+onBeforeUnmount(() => {
+	EventBus.off('route-change', onRouteChange)
+	abortSearch()
+})
+
+const onRouteChange = ({ from, to }: { from: Route, to: Route }): void => {
+	if (to.name !== 'conversation' || from.params.token !== to.params.token || (to.hash && isInCall.value)) {
+		abortSearch()
+		emit('close')
+	}
+}
+
+watch(searchText, (value) => {
+	if (value.trim().length === 0) {
+		searchResults.value = []
+		searchCursor.value = 0
+		isSearchExhausted.value = false
+	}
+})
+
+/**
+ * Cancel search and cleanup the search fields and results.
+ */
+function abortSearch() {
+	cancelSearch.value('canceled')
+	searchText.value = ''
+	fromUser.value = null
+	sinceDate.value = null
+	untilDate.value = null
+	searchDetailsOpened.value = false
+	searchResults.value = []
+	searchCursor.value = 0
+}
+
+/**
+ * Continue fetching more search results
+ */
+function loadMore() {
+	return fetchSearchResults(false)
+}
+
+/**
+ *
+ */
+function fetchNewSearchResult() {
+	return fetchSearchResults(true)
+}
+
+const cancelSearch = ref((cancel: string) => {})
+
+/**
+ * @param [isNew=true] Is it a new search (search parameters changed)?
+ * Fetch the search results from the server
+ */
+async function fetchSearchResults(isNew = true) {
+	// Don't search if the search text is empty
+	if (searchText.value.trim().length === 0) {
+		return
+	}
+
+	isFetchingResults.value = true
+
+	try {
+		cancelSearch.value('canceled')
+		const { request, cancel } = CancelableRequest(searchMessages)
+		cancelSearch.value = cancel
+
+		if (isNew) {
+			searchCursor.value = 0
+		}
+		if (searchCursor.value === 0) {
+			searchResults.value = []
+		}
+
+		const term = searchText.value.trim()
+		if (term.length === 0 && !fromUser.value && !sinceDate.value && !untilDate.value) {
+			return
+		}
+		const response = await request({
+			term: term.length !== 0 ? term : null,
```

Comment:
I don't think we should even search with empty term - but it's already checked above